### ISSUE TITLE
First run always fails: Failed to connect to soffice.bin in 6 seconds.

### ISSUE THREAD

**Issue by akkana** [dagwieers/unoconv#415]
This bug is basically a reboot of bug #219:

The first time I start unoconv after booting, it always fails. I run it (usually from a python script) like this:

unoconv -f html -o /tmp/filename.html /tmp/filename.docx

and the output is:
```
Failed to connect to /usr/lib/libreoffice/program/soffice.bin (pid=1949) in 6 seconds.
Connector : couldn't connect to socket (Success)
Error: Unable to connect or start own listener. Aborting.
```
Subsequent runs work correctly. So I'm guessing it just takes longer than 6 seconds to load all the libraries and start soffice.bin.

I confess that I didn't build unoconv from git (do I have to build all of libreoffice for that?) -- I'm using 0.7-1.1 from the Debian repositories. I see it on my desktop, x86-64 Debian testing
Intel(R) Core(TM) i5 CPU       M 540  @ 2.53GHz
as well as on my slower i386 Debian stable laptop. Both machines have spinning disks, not SSD, which might be a more relevant factor.

I haven't experimented with different -T settings yet.

**Comment by akkana** [comment]
Actually, it doesn't always happen on my netbook (Celeron 847 @ 1.10GHz) even though it's slower, but so far it happens every time on my 64-bit I5 machine. That may explain why I see it more often recently (I converted this machine to 64-bit relatively recently): maybe the larger size of the 64-bit libraries is the limiting factor, not CPU speed.

**Comment by akkana** [comment]
-T 8 doesn't seem to be enough on my 64-bit i5 machine either. Maybe -T 10 will be.

**Comment by tiholic** [comment]
Need help with the same...

**Comment by regebro** [comment]
It's very hard to guess what happens when I can't reproduce this locally. Does increasing -T help? Try a really high number, like 600 (ten minutes).

**Comment by akkana** [comment]
-T 10 has been working for me lately.

**Comment by regebro** [comment]
I don't know why the timeout is so low as 6 seconds by default. I'll keep this open, we should probably change that.

**Comment by LAfricain** [comment]
I have the same issue on Ubuntu 18.04:
```
unoconv -f txt *.htm
Failed to connect to /usr/lib/libreoffice/program/soffice.bin (pid=7140) in 6 seconds.
Connector : couldn't connect to socket (Connection refused)
Error: Unable to connect or start own listener. Aborting.
```

**Comment by amismailz** [comment]
Any one found a solution here?

**Comment by soubhikchatterjee** [comment]
Same problem happening inside docker container

**Comment by regebro** [comment]
What happens if you start soffice by itself?

Can you try with unoserver instead of unoconv?

**Comment by jcomeauictx** [comment]
How about: if a listener is not available at first check, send to stderr something like "No listeners available, launching one now...", and assuming a default timeout of infinity unless one is explicitly stated. Will probably also need to add a `-q`, `--quiet` option to disable this warning.

If this gets lots of upvotes, I may be tempted to implement it and submit a pull request, in my Copious Free Time.

**Comment by regebro** [comment]
It does start a listener by default, and this error is when it for some reason takes so long to start that connecting fails.

**Comment by jcomeauictx** [comment]
What was there in my suggestion that made you surmise I didn't understand that?

**Comment by regebro** [comment]
Yeah, it sounded to me like you suggested starting one if no listener was available. I guess now you are in fact just suggesting to change the default timeout to infinity?

**Comment by jcomeauictx** [comment]
Yes. Two things: the warning, and the default timeout.

**Comment by regebro** [comment]
Yeah, I have no big opinion on that, so that works.

**Comment by dearsina** [comment]
This happens to us on occasion, what seems to resolve the issue is to kill any soffice threads:
```
pkill soffice
```
And then try again. Seems like soffice gets stuck at times.
